src/core/services/compiler/compiler.js

```diff
@@ -6,7 +6,20 @@
  */
 angular
   .module('material.core')
-  .service('$mdCompiler', MdCompilerService);
+  .service('$mdCompiler', MdCompilerService)
+  .config(configureCompiler);
+
+/**
+ * Function that is called on the configuration phase of AngularJS.
+ * It is used to determine if bindings should be pre-assigned or not.
+ */
+function configureCompiler($compileProvider, $provide) {
+  $provide.constant('$$mdPreAssignBindings',
+    // For Angular versions that do not support `preAssignBindingsEnabled`, always pre-assign the
+    // bindings.
+    $compileProvider.preAssignBindingsEnabled ? $compileProvider.preAssignBindingsEnabled() : true
```

Comment:
> Are we sure the `$compileProvider.preAssignBindingsEnabled()` output is set in stone at this point?

As a matter of fact, it is never set in stone :grin: (you can change it any time). Not that this is a common usecase. 

```js
.provider('$$mdPreAssignBindings', function ($compileProvider) {
  var hardcodedValue = (angular.version.major === 1 && angular.version.minor < 6) ? true : false;
  var getValue = $compileProvider.preAssignBindingsEnabled ?
    function () { $compileProvider.preAssignBindingsEnabled(); } :
    function () { return hardcodedValue; };

  this.$get = function () { return getValue; };  
})
```